[FIX]website_sale: fix pricelist dropdown menu position

ktr-odoo opened the pull request:
when pricelist dropdown menu in shop is long beacuse of many pricelists
menu points to top of screen instead of pricelist button

dynamic positioning is used for dropdowns by default. setting display to
static to position menu at proper place

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/69886).

ktr-odoo commented:
@qsm-odoo 
Hi,
I request you to review this PR
Thanks.

JKE-be commented:
hello @ktr-odoo 

Please, before to ask review, try to have green runbot, and commit msg without typo.
You miss a space between [FIX] and module_name, missing Uppercase, dot, ...

Work with draft PR while it is not ready for review...
Once ready, convert it and we got an email. Don't need a second email with your 'custom' msg.

And no opw linked to it ? 

ty & have a great day

ktr-odoo commented:
Hi @JKE-be ,

1) Rebased my branch now runbot is green, updated commit message
2) okay.
3) yes, I found issue while testinng a db in migration, solved here.

Thanks, Have a nice day!!

qsm-odoo commented on addons/website_sale/views/templates.xml:
```diff
@@ -242,7 +242,7 @@
             <t t-set="curr_pl" t-value="website.get_current_pricelist()" />
             <span class="font-weight-bold text-muted">Pricelist:</span>
-            <a role="button" href="#" class="dropdown-toggle btn btn-light border-0 px-0 text-muted align-baseline" data-toggle="dropdown">
+            <a role="button" href="#" class="dropdown-toggle btn btn-light border-0 px-0 text-muted align-baseline" data-toggle="dropdown" data-display="static">
```
By doing this, you create other bugs, like this one (especially true in mobile):
![Screenshot from 2021-04-28 10-33-30](https://user-images.githubusercontent.com/10338094/116373035-46c09700-a80d-11eb-966f-ed600ec0fa4e.png)

I guess this could be fixed by aligning the dropdown-menu on the right with the `dropdown-menu-right` class but I wonder if the current "bug" is a real one worth fixing at all? And if needed, is not it to be fixed in stable versions as well?

ktr-odoo replied:
Hi, @qsm-odoo 
Thanks for inputs. :)
indeed this fixed issue (https://drive.google.com/file/d/1rCpQAF4ZsLsMDKC_TIW5swHBK-tCw-JF/view?usp=sharing)

please check the below screenshot for current bug,
https://drive.google.com/file/d/1z5v0kFzzSoBQpuQWU5f2lagx_5x0SQY_/view?usp=sharing
when list is too long due to dynamic postioining of dropdown. it attaches to top of page (above pricelist button).

Yes, this is problem from the version 12. so, to be fixed in stable versions.
But as the change is made in python template I have targeted master.
